app/gcp/instancemanager.go

```diff
@@ -104,6 +104,39 @@ func (m *InstanceManager) CreateHost(zone string, req *apiv1.CreateHostRequest,
 	return result, nil
 }
 
+const listHostsRequestMaxResultsLimit = 500
+
+func (m *InstanceManager) ListHosts(zone string, user app.UserInfo, req *app.ListHostsRequest) (*apiv1.ListHostsResponse, error) {
+	if req.MaxResults > listHostsRequestMaxResultsLimit {
```

Comment:
this doesn't have to be an error, you can return at most listHostsRequestMaxResultsLimit elements and a nextPageToken instead.

Reply:
done